saleor/graphql/product/mutations/products.py

```diff
@@ -1785,6 +1788,54 @@ def perform_mutation(cls, _root, info, product_id, images_ids):
         return ProductImageReorder(product=product, images=images)
 
 
+class ProductVariantSetDefault(BaseMutation):
+    product = graphene.Field(Product)
+
+    class Arguments:
+        product_id = graphene.ID(
+            required=True,
+            description="Id of a product that will have the default variant set.",
+        )
+        variant_id = graphene.ID(
+            required=True, description="Id of a variant that will be set as default.",
+        )
+
+    class Meta:
+        description = (
+            "Set default variant for a product. "
+            "Mutation triggers PRODUCT_UPDATED webhook."
+        )
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductError
+        error_type_field = "product_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, product_id, variant_id):
+        product = cls.get_node_or_error(
+            info, product_id, field="product_id", only_type=Product
+        )
+        variant = cls.get_node_or_error(
+            info,
+            variant_id,
+            field="variant_id",
+            only_type=ProductVariant,
+            qs=models.ProductVariant.objects.select_related("product"),
+        )
+        if variant.product != product:
```

Comment:
Maby?
```suggestion
            only_type=ProductVariant,
        )
        if variant.product_id != product.id:
```

Reply:
the qs saves one query as we compare resulting related object to one we already have, measured with `query_count`.